balboa-core/src/main/scala/com/socrata/balboa/metrics/data/impl/Cassandra11QueryImpl.scala

```diff
@@ -88,30 +93,39 @@ class Cassandra11QueryImpl(context: AstyanaxContext[Keyspace]) extends Cassandra
     logger debug s"Using entity/row key $entityKey at period $period"
     fastfail.proceedOrThrow()
 
-
-    val m:MutationBatch = context.getEntity.prepareMutationBatch
-      .setConsistencyLevel(ConsistencyLevel.CL_ONE)
-      .withRetryPolicy(new ExponentialBackoff(250, 5))
-    if (aggregates.nonEmpty) {
-      var cols = m.withRow(Cassandra11Util.getColumnFamily(period, RecordType.AGGREGATE), entityKey)
-      for { (k,v) <- aggregates } {
-        if (k != ""){ cols = cols.incrementCounterColumn(k, v.getValue.longValue) }
-        else { logger warn("dropping metric with empty string as column") }
-      }
-    }
-
-    if (absolutes.nonEmpty) {
-      var cols = m.withRow(Cassandra11Util.getColumnFamily(period, RecordType.ABSOLUTE), entityKey)
-      for { (k,v) <- absolutes } {
-        if (k != ""){ cols = cols.putColumn(k, v.getValue.longValue) }
-        else { logger warn("dropping metric with empty string as column") }
+    val batchStatement = new BatchStatement(BatchStatement.Type.LOGGED)
+
+    val entityKeyWhere = QueryBuilder.eq("key", entityKey)
+
+    for (sameTypeRecords <- List(aggregates, absolutes)) {
```

Comment:
How about

    for (sameTypeRecords <- List(aggregates, absolutes).filter(_.nonEmpty)) {

and removing a layer of nesting from the loop. Would that work?